src/main/webapp/script.js

```diff
@@ -117,3 +140,17 @@ function deleteTask(task) {
   params.append('id', task.id);
   fetch('/delete-task', {method: 'POST', body: params});
 }
+
+/** Tells the server to delete the grocery. */
+function deleteGrocery(grocery) {
+  const params = new URLSearchParams();
+  params.append('id', grocery.id);  
+  fetch('/delete-grocery', {method: 'POST', body: params});
+}
+
+function userMatch(grocery) {
+  if (grocery.userMatch === true) {
```

Comment:
`userMatch` function should be changed to something like `isEditableGrocery`, maybe?

Based on previous suggested changes, I would suggest

`function isEditable(grocery) {
  return grocery.isAssignedToUser || !grocery.assignee
}`

If you're returning a boolean value, don't use conditions (unless it's super long and confusing) -- try to see if you can fit it in a single line

Reply:
solved